backend/tests/v5/unit/processors/users.test.js

```diff
@@ -0,0 +1,39 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { src } = require('../../helper/path');
+
+const Users = require(`${src}/processors/users`);
+
+jest.mock('../../../../src/v5/models/users');
+const UsersModel = require(`${src}/models/users`);
+
+UsersModel.canLogIn.mockImplementation((user) => user);
+UsersModel.authenticate.mockResolvedValue('user1');
+
+const testLogin = () => {
+	describe('Login', () => {
+		test('should login with username', async () => {
```

Comment:
Can we add a test where it fails -> on either canLogIn or authenticate, to make sure it passes on the error codes